### DISCUSSION TITLE
support typescript option `noUncheckedSideEffectImports`

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#70165] [Ideas]
### Link to the code that reproduces this issue

https://github.com/stefanprobst/issue-next-no-unchecked-side-effect-imports

### To Reproduce

1. clone [repo](https://github.com/stefanprobst/issue-next-no-unchecked-side-effect-imports) and install dependencies
2. `pnpm tsc --noEmit`
3. see error message:

```
app/layout.tsx:3:8 - error TS2307: Cannot find module './globals.css' or its corresponding type declarations.

3 import "./globals.css";
```

### Current vs. Expected behavior

should support typescript option `noUncheckedSideEffectImports` (introduced in v5.6)

probably needs `.d.ts` for `.css` imports

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #202403151937 SMP PREEMPT_DYNAMIC Fri Mar 15 19:52:22 UTC 2024
  Available memory (MB): 31401
  Available CPU cores: 16
Binaries:
  Node: 22.8.0
  npm: 10.8.2
  Yarn: 1.22.22
  pnpm: 9.9.0
Relevant Packages:
  next: 14.2.11 // Latest available version is detected (14.2.11).
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.6.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

create-next-app, TypeScript

### Which stage(s) are affected? (Select all that apply)

next build (local)

**Comment by samcx** [comment]
@stefanprobst Since this isn't really a 🐛 with Next.js, I have converted this to a discussion.

You can also just add the needed file.

![CleanShot 2024-09-16 at 16 00 03@2x](https://github.com/user-attachments/assets/b0e43c7d-b4c6-45d5-bde6-d28d6ba31f8c)

**Reply by stefanprobst** [reply]
does it make sense to include this with the autogenerated `next-env.d.ts`?

**Reply by samcx** [reply]
@stefanprobst Possibly in the future! But since it's a brand new config, we should leave it for now until it gains more traction.

**Reply by Janpot** [reply]
@samcx FYI: `noUncheckedSideEffectImports` will become default `true` in upcoming typescript [major](https://github.com/microsoft/TypeScript/issues/54500). As such I propose to reopen the [issue](https://github.com/vercel/next.js/issues/70106) and include the ambient declaration for css files in next.js types.

**Reply by samcx** [reply]
@Janpot Unfortunately, it's not possible to re-open (I think because it was converted to a discussion). Can you push up a new issue outlining this?

**Reply by Janpot** [reply]
Sure https://github.com/vercel/next.js/issues/88197

**Comment by MuhammadAflah** [comment]
If doing the same for react vite app will resolve the "Unknown compiler option 'noUncheckedSideEffectImports'." issue ?

**Reply by Sanjuchilukuri** [reply]
Hi,
@MuhammadAflah, is that issue resolved?

**Comment by Netail** [comment]
Has been fixed in NextJS 16.2
